### ISSUE TITLE
Atom stalls when Julia is started

### ISSUE THREAD

**Issue by robinkim28** [JunoLab/atom-julia-client#405]
Hi,
 I installed Atom and uber-juno package after installing Julia 0.6.0 on my machine, and Atom cannot not process any Julia code. Julia is working through terminal and ipython notebook, but just not through Atom.

When I click Julia > Start Julia, a progress bar in the bottom left corner of Atom just keeps going until I manually hit Julia > Stop Julia again. Similarly, when I open up a new .jl file and type any command and hit Shift+Enter, same thing happens again.

Any help would be appreciated.

**Comment by pfitzseb** [comment]
Can you open the Console (Packages > Julia > Open Console or <kbd>Ctrl-J Ctrl-O</kbd>) and paste any errors you see here?

**Comment by robinkim28** [comment]
Console opens up and when I type anything and hit Enter, it again shows the bottom left corner's progress bar revolving for indefinite amount of time. See the bottom left corner of the screen shot. Even if I type gibberish, it spins without outputting any message. 

![screen shot 2018-01-12 at 5 29 54 pm](https://user-images.githubusercontent.com/17394971/34897857-4796f51e-f7be-11e7-9300-fa985086467b.png)

**Comment by pfitzseb** [comment]
Can you go to `View > Developer > Toggle Developer Tools` and report any errors you see there? 
Please also check that the path you've set in `Settings > Packages > julia-client > Settings > Julia Path` is correct and works when you type it into a terminal of your choice.
Lastly, what OS are you on?

**Comment by robinkim28** [comment]
I am not sure what kind of error messages are outputted from the developer window. Which tab should I focus on out of Elements, Console, Sources, Network, Timeline, Profiles, Applications, Security and Audits?

Under Console, I see these three messages:
<embedded>:121025 IntersectionObserver.observe(target): target element is not a descendant of root.
<embedded>:1042 EACCES: permission denied, symlink '/Applications/Atom.app/Contents/Resources/app/apm/node_modules/.bin/apm' -> '/usr/local/bin/apm'
<embedded>:1037 EACCES: permission denied, symlink '/Applications/Atom.app/Contents/Resources/app/atom.sh' -> '/usr/local/bin/atom'

Julia Path is set to default:julia and when I type julia from terminal, it loads up Julia.

I am on Mac OS Sierra 10.12.6.

**Comment by pfitzseb** [comment]
Alright, thanks for the info. Can you type `using Atom` into a julia terminal (outside of Atom) and report any errors you might see?

**Comment by robinkim28** [comment]
Sure! This might be the problem?

julia> using Atom
ERROR: ArgumentError: Module Atom not found in current path.
Run `Pkg.add("Atom")` to install the Atom package.
Stacktrace:
 [1] _require(::Symbol) at ./loading.jl:435
 [2] require(::Symbol) at ./loading.jl:405

**Comment by pfitzseb** [comment]
Yes, that does look like the problem. Can you try again after running `Pkg.add("Atom")` as suggested?

**Comment by robinkim28** [comment]
Here is the message. The issue hasn't gone away within the Atom client. The progress bar still stalls.. 

julia> using Atom
INFO: Precompiling module Atom.

WARNING: deprecated syntax "inner constructor Client(...) around /Users/robinkim/.julia/v0.6/HttpServer/src/HttpServer.jl:114".
Use "Client{T}(...) where T" instead.
WARNING: using Lazy.remove in module AtomShell conflicts with an existing identifier.

**Comment by pfitzseb** [comment]
Please try `Pkg.build("Atom")` as well.

**Comment by Cortalak** [comment]
Hi, I experienced the same problem on windows10 with Julia version 0.6.2. When I'm invited to press Enter to start Julia in the console, Julia didn't start and the progress bar runs forever. I tried on a diffrent machine and had the same results. Also tried with Juno that comes with JuliaPro and again had the same problem. However I was able to make Julia run by disabling the Powershell Wrapper in Julia-Client.

**Comment by xanfus** [comment]
Boot mode "cycler", console is REPL-based. Devtools report, WinPTY agent doesn't exist within julia-client package.

**Comment by pfitzseb** [comment]
That sounds like a messed up installation -- can you try reinstalling? Also maybe disable your virus scanner temporarily and install julia client via `apm install julia-client` in a terminal.

**Comment by daniesso** [comment]
Hi. Has there been found any resolutions for this problem? We've adopted Julia for our Algorithms and Datastructures course, and recommend Atom/Juno as IDE. But a lot of students with Windows computers have this problem. 

Reinstalls doesn't fix it, nor does checking the path. They all have updated versions, including julia 1.0.0.

**Comment by daniesso** [comment]
Ah. No, the main issue, with the REPL hanging on "Press Enter to start Julia."

**Comment by pfitzseb** [comment]
I have never been able to repro that issue, so a couple of questions:

Does `using Pkg; Pkg.add("Atom"); using Atom` work in a REPL? If you open the devtools with `Ctrl-Shift-I`, are there any errors?

**Comment by daniesso** [comment]
`using Pkg; Pkg.add("Atom");` gave an error + stacktrace starting with
 ```
ERROR: The following package names could not be resolved:
* Atom (not found in project, manifest or registry)
Please specify by known `name=uuid`.
```

**Comment by daniesso** [comment]
Another student proposed the following, which solved the problem for both of them.
```
rm(joinpath(homedir(), ".julia", "registries"); recursive=true)

using Pkg; Pkg.add("Atom"); using Atom
```

**Comment by pfitzseb** [comment]
Ah, right, I think that's a Pkg bug that'll be fixed in Julia 1.0.1. Glad you figured it out! :)
